emission/tests/exportTests/TestExportPipelineConfirmedTripsandBackground.py

```diff
@@ -0,0 +1,75 @@
+import emission.storage.timeseries.abstract_timeseries as esta
+import gzip
+import glob
+import emission.tests.common as etc
+import emission.pipeline.export_stage as epe
+import emission.storage.pipeline_queries as espq
+import emission.exportdata.export_data as eeed
+
+class TestExportPipelineConfirmedTripsandBackground(unittest.TestCase):
+    def testExportPipeline(self):
+        etc.setupRealExample(self, "emission/tests/data/real_examples/shankari_2015-07-22")
+        etc.runIntakePipeline(self.testUUID)
+        eeed.export_data(self.testUUID)
+        
+        #Data has been run through both pipelines, file has been exported.
+        tq = espq.get_time_range_for_export_data(self.testUUID)
+        exported_file_name = "emission/archived/archive_%s_%s_%s.gz" % (self.testUUID, tq.startTs, tq.endTs)
+        
+        #trim this file name from the back 
+        #generate a list of filesnames that are close (exports that are recent with this user, small list)
+	#this is a HACK, is there a better way? I am lost!(hoping list is 1)
+        
+        exported_file_name_trimmed = exported_file_name[:len(exported_file_name) - 15]
+        file_list = []
+        for name in glob.glob("%s*.gz" % exported_file_name_trimmed):
+            file_list.append(name)
+        for exported_file_name in file_list:
+            with gzip.open(exported_file_name, 'r') as ef:
+                exported_data = json.loads(ef.read().decode('utf-8'))
+            
+            #Testing to confirm we have confirmed trips in the exported data, none in raw, should not be equal
+
+            confirmed_trips_exported = []
+            for t in exported_data:
+                if t['metadata']['key'] == "analysis/confirmed_trip":
+                    confirmed_trips_exported.append(t)
+            raw_data = self.readDataFromFile("emission/tests/data/real_examples/shankari_2015-07-22")
+            confirmed_trips_raw = []
+            for t in raw_data:
+                if t['metadata']['key'] == "analysis/confirmed_trip":
+                    confirmed_trips_raw.append(t)
+            self.assertNotEqual(confirmed_trips_exported, confirmed_trips_raw)
+
+            #Testing to confirm there is no confirmed trips in the raw data (validates prev test)
+
+            empty_list = []
+            self.assertEqual(empty_list, confirmed_trips_raw)
+
+            #Testing to confirm the length of data entries, should be the same for raw and processed
+
+            background_location_exported = []
+            for t in exported_data:
+                if t['metadata']['key'] == "background/location":
+                    background_location_exported.append(t)
+            background_location_raw = []
+            for t in raw_data:
+                if t['metadata']['key'] == "background/location":
+                    background_location_raw.append(t)
+            self.assertEqual(len(background_location_exported), len(background_location_raw)) 
```

Comment:
- In addition to asserting on the length, maybe assert on the first few entries.
- also, this is checking raw data, you also want to check some analysis results

Reply:
As replied below as well, I think I am having some confusion with accessing what is in the database and being able to compare that with what is in the export file.

Reply:
Not sure what the confusion is:
- you read the entries from the database using the timeseries, keys and timequery
- you read the export file and the entries through a file read

you compare them.

Can you clarify your confusion?

Reply:
This makes sense. Before I was struggling to have access to the time_query and timeseries structure because I was calling the pipeline from too high of a level.

Reply:
>In addition to asserting on the length, maybe assert on the first few entries.

`First differing element 0:
{'_id': {'$oid': '60e5e60fe9a59284f8e0e2a7'}, 'us[1321 chars] {}}}
{'_id': ObjectId('60e5e60fe9a59284f8e0e2a7'), 'us[1320 chars] {}}}
`

Something I am seeing when attempting to do the assert calls is a slight difference in formatting/character choice. I attempted to use the assertAlmostEqual to tap around this, however it seems to only work on listed values of numerical value only. Is there a better way to test this?

Reply:
this is because of the bson dump process (bson.objectid). Maybe get out the object id string (`trip["_id"]["$oid"]`) and compare that directly? you will have to do something similar for the UUID as well.

Reply:
`self.assertEqual(db_trip_one['user_id'],exported_trip_one['user_id']['$uuid'])
AssertionError: UUID('560c4dff-3485-4a3a-9e52-ac2d0a8c3c9f') != '560c4dff34854a3a9e52ac2d0a8c3c9f'

Hmm, is there a reason I am unable to index into the UUID? It does not seem to be in the typical dictionary format. Is there a way around this? Or indexing techniques into the bson dump format?

Reply:
Same issue with the ObjectId, looking to just pull out the 128 bit sequence.

Reply:
these are objects, please look at their interfaces. You can't assume that an object will match a string.

Reply:
I see the it is an object, but am having a hard time finding any documentation on how to access the object.